src/main/java/hudson/remoting/Engine.java

```diff
@@ -184,13 +179,13 @@ public void run() {
 
                     // find out the TCP port
                     HttpURLConnection con = (HttpURLConnection)salURL.openConnection();
-                    if (con instanceof HttpURLConnection) {
+                    if (con != null) {
```

Comment:
Obviously the previous code was wrong (doing a `checkcast` before `instanceof`), but this is probably wrong too; AFAIK `openConnection` cannot return null.